pkg/datapath/linux/node.go

```diff
@@ -566,15 +566,14 @@ func upsertIPsecLog(err error, spec string, loc, rem *net.IPNet, spi uint8) erro
 
 func (n *linuxNodeHandler) registerIpsecMetricOnce() {
 	n.ipsecMetricOnce.Do(func() {
-		metrics.Register(n.ipsecMetricCollector)
+		if err := metrics.Register(n.ipsecMetricCollector); err != nil {
+			log.WithError(err).Errorf("IPSec metrics registration failed. No metrics will be reported!")
+		}
```

Comment:
This metrics registration code is a bit interesting; I don't think it is possible for it to return errors, **currently**. Looking at the [code](https://github.com/cilium/cilium/blob/d06140fc9e604a39c6cf71ee596cbb0e27691ef3/pkg/metrics/metrics.go#L1432-L1438), the error is never captured. We should probably fix this, cc @dylandreimerink as last author of the code. Once it's captured, then the changes in the PR would actually be utilized. 

I'm also wondering what sort of errors metrics registration could return, but I digress...

For this PR, my only comment is we don't need the dynanmic formatting:

```suggestion
		if err := metrics.Register(n.ipsecMetricCollector); err != nil {
			log.WithError(err).Errorf("IPsec metrics registration failed. No metrics will be reported!")
		}
```

Reply:
Right noticed that too. But figure I'll flag it. Since we swallow the error too in the wrapper??

Reply:
Agreed, that just ignoring the errors is not the correct way to deal with that. I don't exactly know why I made that decision. I think I did so because ultimately the goal is to migrate all metrics registration over to using hive instead of these global methods, which would move any error reporting burden to the registry anyway. But until then, we should at least be aware of registration errors.

I think the code is good as is, given the function signature has an error, it should always be checked in case the function starts returning actual errors.